app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,157 @@
+class AccountsController < ApplicationController
+  before_action :set_account, only: [:show, :edit, :update, :destroy, :do_withdraw, :do_deposit, :deposit, :withdraw, :verify_pin, :send_money, :change_status, :check_pin]
+  before_action :authorize_account, only: [:show, :edit, :update, :destroy, :do_withdraw, :do_deposit, :deposit, :withdraw, :verify_pin, :send_money, :change_status, :check_pin]
+  def index
+    @accounts = current_user.accounts
+  end
+
+  def new
+    account = Account.order(:account_number).last
+    next_number = account ? account.account_number.to_i + 1 : 0
+    @account = current_user.accounts.build(account_number: next_number)
+  end
+
+  def create
+    @account = current_user.accounts.build(account_params)
+    if @account.save
+      redirect_to @account
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    @account.update(account_params) ? redirect_to(@account) : render(:edit, status: :unprocessable_entity)
+  end
+
+  def destroy
+    @account.delete
+    redirect_to accounts_path, notice: 'Account deleted successfully'
+  end
+
+  def do_withdraw
+    amount = get_amount
+    if sufficient_balance?(amount)
+      @account.update(balance: @account.balance - amount)
+      create_transaction(amount, 'withdraw')
```

Comment:
Shouldn't this be in a transaction? without transaction you'll eventually get data inconsistency.